eve/main.yml

```diff
@@ -589,6 +591,83 @@ models:
       alwaysRun: true
       sigtermTime: 600
 
+  # --- Solutions management ---
+  - ShellCommand: &retrieve_iso_solution
+      # NOTE: Retrieve solution from property
+      # "<solution_base_url>/<solution_archive>"
+      # OR if solution_base_url not set
+      # "<solution_repo>/<solution_version>/<solution_archive>"
+      # default to:
+      # "https://github.com/scality/metalk8s-solution-example/releases/download/1.0.0/example-solution-1.0.0.iso"
+      <<: *retrieve_iso
+      doStepIf: "%(prop:install_solution:-false)s"
+      name: Retrieve Solution archive
+      env:
+        <<: *_env_retrieve_artifact_retry
+        BASE_URL: >-
+          %(prop:solution_base_url:-%(prop:solution_repo:-https://github.com/scality/metalk8s-solution-example/releases/download)s/%(prop:solution_version:-1.0.0)s)s
+        FILE_SOURCE: "%(prop:solution_archive:-example-solution-1.0.0.iso)s"
+        FILE_DEST: "%(prop:solution_archive:-example-solution-1.0.0.iso)s"
+  - ShellCommand: &copy_solution_archive_bootstrap_ssh
+      doStepIf: "%(prop:install_solution:-false)s"
+      name: Copy Solution archive to bootstrap
+      <<: *copy_iso_bootstrap_ssh
+      env:
+        <<: *_env_copy_iso_bootstrap_ssh
+        ARCHIVE: "%(prop:solution_archive:-example-solution-1.0.0.iso)s"
+  - ShellCommand: &import_solution
+      doStepIf: "%(prop:install_solution:-false)s"
+      name: Import Solution archive
+      env:
+        SOLUTION_ARCHIVE: "%(prop:solution_archive:-example-solution-1.0.0.iso)s"
+        METALK8S_MOUNTPOINT: "/srv/scality/metalk8s-%(prop:metalk8s_version)s"
+      command: >
+        ssh -F ssh_config bootstrap
+        sudo "$METALK8S_MOUNTPOINT/solutions.sh"
+        import --archive "\$(readlink -f \"$SOLUTION_ARCHIVE\")"
+      workdir: build/eve/workers/openstack-terraform/terraform/
+      haltOnFailure: true
+  - ShellCommand: &activate_solution
+      doStepIf: "%(prop:install_solution:-false)s"
+      name: Activate Solution
+      env:
+        SOLUTION_NAME: "%(prop:solution_name:-example-solution)s"
+        SOLUTION_VERSION: "%(prop:solution_version:-1.0.0)s"
```

Comment:
Can't we read these from the downloaded archive? This could serve as a minimal check, and would reduce the number of build props to set (using a Zenko archive as an example). The current approach needs at least 5 properties to be set:

  - `install_solution=true`
  - `solution_base_url=https://zenko-artifacts/2.3.4/` (or `solution_repo=https://zenko-artifacts/`)
  - `solution_archive=zenko-2.3.4.iso`
  - `solution_name=zenko`
  - `solution_version=2.3.4`

Reading the name and version from the ISO could reduce the required props to `install_solution`, `solution_base_url` and `solution_archive` (the two latter could be reduced to a single `solution_archive_url` from which we can extract the basename).

Reply:
Hmm you right, I will do it, in another PR